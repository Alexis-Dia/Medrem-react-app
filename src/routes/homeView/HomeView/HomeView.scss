.home-view-container {
  margin: 14% 0 0 0 !important;
  width: 100% !important;
}

.home-view-paper {
  min-width: 250px;
  min-height: 250px;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: space-around;

  &:hover {
    color: #3f51b5;
    cursor: pointer;

    img {
      transform: scale(1.25);
    }
  }

  img {
    transition: all 0.8s ease;
  }
}

.home-menu-icon {
  width: 70px;
  height: 70px;
}