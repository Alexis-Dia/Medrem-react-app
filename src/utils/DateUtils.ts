class DateUtils {

    static readonly DATE_TIME_MASK = [
        /\d/,
        /\d/,
        /\d/,
        /\d/,
        "/",
        /\d/,
        /\d/,
        "/",
        /\d/,
        /\d/,
        " ",
        /\d/,
        /\d/,
        ":",
        /\d/,
        /\d/,
    ];

    static readonly DATE_TIME_FORMAT_DEFAULT = "YYYY/MM/DD HH:mm";
    static readonly DATE_YY_MM_DD_FORMAT = "yy/MM/dd";

    private static readonly VALID_PARTS_LENGTH = 4;

    static parseDate = (dateStr: string, format: string): Date | null => {
        const regex = format.toLocaleLowerCase()
            .replace(/\bd+\b/, "(?<day>\\d+)")
            .replace(/\bm+\b/, "(?<month>\\d+)")
            .replace(/\by+\b/, "(?<year>\\d+)");

        const parts = new RegExp(regex).exec(dateStr);
        if (parts) {
            //@ts-ignore
            const { year, month, day } = parts.groups;
            return parts.length === DateUtils.VALID_PARTS_LENGTH ? new Date(year, month - 1, day) : null;
        }
        return null;
    }
}

export default DateUtils;
