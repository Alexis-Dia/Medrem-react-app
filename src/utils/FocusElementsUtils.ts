class FocusElementsUtils {

    private static getComboBox = (element: any): HTMLElement | null => {

        for (const childNode of element.childNodes) {
            let attribute;
            if (childNode instanceof Element) {
                attribute = childNode.getAttribute("aria-haspopup");
            }
            if (attribute) {
                return childNode;
            } else {
                const comboBox = FocusElementsUtils.getComboBox(childNode);
                if (comboBox) {
                    return comboBox;
                }
            }
        }
        return null;
    }

    private static getInputOrTextarea = (element: any): HTMLInputElement | null => {

        const inputs = element.getElementsByTagName("input");
        if (inputs && inputs.length > 0) {
            return inputs[0];
        }
        const textareas = element.getElementsByTagName("textarea");
        if (textareas && textareas.length > 0) {
            for (const textarea of textareas) {
                if (textarea.getAttribute("aria-invalid")) {
                    return textarea;
                }
            }
            return textareas[textareas.length - 1];
        }
        return null;
    }

    private static getButton = (element: any): HTMLElement | null => {

        const buttons = element.getElementsByTagName("button");
        if (!buttons) {
            return null;
        }

        return buttons[0];
    }

}

export default FocusElementsUtils;
