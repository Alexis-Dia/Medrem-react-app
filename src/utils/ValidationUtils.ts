import { ValidationRule } from "../types/validation";

class ValidationUtils {

    static getRuleByFieldName = (fieldName: string): ValidationRule => {
        switch (fieldName) {
            case "email":
                return ValidationRule.EMAIL;
            case "password":
                return ValidationRule.PASSWORD;
            case "username":
                return ValidationRule.USERNAME;
            case "firstName":
            case "lastName":
                return ValidationRule.VALUE_REQUIRED;
            case "externalEmployeeId":
                return ValidationRule.EMPLOYEE;
            default:
                return ValidationRule.VALUE_REQUIRED;
        }
    };

}

export default ValidationUtils;
