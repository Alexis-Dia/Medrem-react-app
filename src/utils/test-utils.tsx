/* istanbul ignore file */

import React from "react";
import { IntlProvider } from "react-intl";
import { Provider } from "react-redux";
import { MuiThemeProvider } from "material-ui/styles";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

export enum Wrapper {
    REDUX_PROVIDER,
    INTL_PROVIDER,
    THEME_PROVIDER
}

export const getWrappedComponent = (component: JSX.Element, wrappers: Array<Wrapper>, messages = {}, store = {}): JSX.Element => {
    let wrappedComponent = component;
    wrappers.forEach((wrapper) => {
        wrappedComponent = wrapComponent(wrappedComponent, wrapper, messages, store);
    });
    return wrappedComponent;
};

const wrapComponent = (component: JSX.Element, wrapper: Wrapper, messages: any, store: any) => {
    switch (wrapper) {
        case Wrapper.INTL_PROVIDER:
            return injectIntl(component, messages);
        case Wrapper.REDUX_PROVIDER:
            return injectStore(component, store);
        case Wrapper.THEME_PROVIDER:
            return injectStyles(component);
        default:
            return component;
    }
};

const injectStore = (node: JSX.Element, store: any): JSX.Element => (
    <Provider store={store}>
        {node}
    </Provider>
);

const injectIntl = (node: JSX.Element, messages: any): JSX.Element => (
    <IntlProvider locale="en" defaultLocale="en" messages={messages}>
        {node}
    </IntlProvider>
);

const injectStyles = (node: JSX.Element): JSX.Element => (
    <MuiThemeProvider>
        {node}
    </MuiThemeProvider>
);

export const findByAriaLabel = (component: any, ariaLabel: string) => component.find(`[aria-label='${ariaLabel}']`);

interface MountingParams {
    component: JSX.Element;
    name: string;
}

export const mountComponent = (params: MountingParams) => {
    const wrapper = mount(params.component);
    return {
        wrapper,
        component: wrapper.find(params.name),
    };
};

export const renderComponent = (params: MountingParams) => renderer.create(params.component);
