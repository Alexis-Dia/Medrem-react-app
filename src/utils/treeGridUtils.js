export function getTreeArr(arr) {
    const newArr = [];
    let mainGeneratedId = 0;
    let generatedId = 100;
    arr.forEach((item) => {
        Array.isArray(newArr)&& newArr.push({
            "parentId": null,
            "id": mainGeneratedId,
            "name": item.name,
        });
        mainGeneratedId = mainGeneratedId + 1;
        let currentId = 0;
        if (Array.isArray(item.fields)&& item.fields.length > 0) {
            currentId = mainGeneratedId - 1;
        }
        Array.isArray(item.fields)&& item.fields.forEach((part) => {
            newArr.push({
                "parentId": currentId,
                "id": mainGeneratedId,
                "name": part.name,
                "parentName": item.name,
                "selectedByDefault": part.selectedByDefault,
            });
            mainGeneratedId = mainGeneratedId + 1;
            generatedId = generatedId + 1;
        });
    });
    generatedId = 0;
    return newArr;
}
