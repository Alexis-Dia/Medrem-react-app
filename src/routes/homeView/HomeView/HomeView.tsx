import React from "react";
import { connect, ConnectedProps } from "react-redux";
import { bindActionCreators } from "redux";
import { injectIntl, FormattedMessage, WrappedComponentProps } from "react-intl";
import { Grid, Paper } from "@material-ui/core";
import UsersIcon from "../images/multiple-users-silhouette.png";
import OffersIcon from "../images/case.png";
import PermissionIcon from "../images/cubes.png";
import KpiIcon from "../images/statistic.png";
import ScmObjectIcon from "../images/scm-object.png";
import browserHistory from "../../history";
import { CommonUtils } from "../../../utils";
import PublishedSystemInformation from "../PublishedSystemInformation/PublishedSystemInformation";
import "./HomeView.scss";
import { AppPath } from "../../../properties/appPath";

type Props = PropsFromRedux & WrappedComponentProps

class HomeView extends React.Component<Props> {
    componentDidMount() {
    }

	handleKpiClick = () => {
	    browserHistory.push(AppPath.KPI);
	};

	handleUsersManagementClick = () => {
	    browserHistory.push(AppPath.USERS);
	};

	handleOffersManagementClick = () => {
	    browserHistory.push(AppPath.DOCUMENTS);
	};

	handleRolesManagementClick = () => {
	    browserHistory.push(AppPath.ROLES);
	};

	handleScmObjectClick = () => {
	    browserHistory.push(AppPath.SCM_OBJECTS);
	}

	render = () => {
	    const allFeatures = localStorage.getItem("features");

	    return (
	        <div>aaa</div>
	    );
	};
}

const mapDispatchToProps = (dispatch: any) => ({

});

const connector = connect(null, mapDispatchToProps);

type PropsFromRedux = ConnectedProps<typeof connector>;

export default injectIntl(connector(HomeView));
