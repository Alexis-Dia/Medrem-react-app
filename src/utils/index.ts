import ValidationUtils from "./ValidationUtils";

export {
    ValidationUtils,
};
