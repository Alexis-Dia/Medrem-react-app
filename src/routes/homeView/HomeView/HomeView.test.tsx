import React from "react"
import HomeView from "./HomeView"
import configureStore from "redux-mock-store";
import { getWrappedComponent, mountComponent, renderComponent, Wrapper } from "../../../utils/test-utils";
import history from "../../history";
import { CommonUtils } from "../../../utils";

const mockStore = configureStore();
const store = mockStore({});

const getWrappedComponentImpl = () => {
    const wrappers = [Wrapper.REDUX_PROVIDER, Wrapper.INTL_PROVIDER];
    const component = <HomeView />;
    return { component: getWrappedComponent(component, wrappers, {}, store), name: "HomeView" };
};

const getFeaturesMock = CommonUtils.ifFeaturesExists = jest.fn().mockImplementation(() => true);

jest.mock("material-ui-old-core/Grid", () => "grid-mock");
jest.mock("material-ui-old-core/Paper", () => "paper-mock");
jest.mock("../PublishedSystemInformation/PublishedSystemInformation", () => "published-system-info-mock");
jest.mock("../../history", () => ({
    push: jest.fn(),
}));

describe("<HomeView />", () => {
    const { wrapper: wrapperWithInitialProps } = mountComponent(getWrappedComponentImpl());

    afterEach(() => {
        getFeaturesMock.mockClear();
        //@ts-ignore
        history.push.mockClear();
    });

    it("renders correctly", () => {
        const tree = renderComponent(getWrappedComponentImpl());
        expect(tree).toMatchSnapshot();
    });

    describe("actions simulation", () => {
        const gridItems = wrapperWithInitialProps.findWhere((grid) => grid.props().item === true && !grid.props().xs);
        expect(gridItems).toHaveLength(4);

        it("clicks on kpi item", () => {
            gridItems.at(0).props().onClick();
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith("/kpi");
        });

        it("clicks on users item", () => {
            gridItems.at(1).props().onClick();
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith("/users");
        });

        it("clicks on offers item", () => {
            gridItems.at(2).props().onClick();
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith("/documents/");
        });

        it("clicks on permissions item", () => {
            gridItems.at(3).props().onClick();
            expect(history.push).toBeCalledTimes(1);
            expect(history.push).toBeCalledWith("/roles");
        });
    });
});
