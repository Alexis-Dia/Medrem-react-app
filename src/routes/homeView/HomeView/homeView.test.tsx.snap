// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeView /> renders correctly 1`] = `
<div>
  <grid-mock
    className="home-view-container"
    container={true}
    spacing={16}
  >
    <grid-mock
      item={true}
      xs={12}
    >
      <grid-mock
        container={true}
        justify="center"
        spacing={16}
      >
        <grid-mock
          item={true}
          onClick={[Function]}
        >
          <paper-mock
            className="home-view-paper"
          >
            <img
              className="home-menu-icon"
              src="test-file-stub"
            />
            <h6>
              KPI
            </h6>
          </paper-mock>
        </grid-mock>
        <grid-mock
          item={true}
          onClick={[Function]}
        >
          <paper-mock
            className="home-view-paper"
          >
            <img
              className="home-menu-icon"
              src="test-file-stub"
            />
            <h6>
              User management
            </h6>
          </paper-mock>
        </grid-mock>
        <grid-mock
          item={true}
          onClick={[Function]}
        >
          <paper-mock
            className="home-view-paper"
          >
            <img
              className="home-menu-icon"
              src="test-file-stub"
            />
            <h6>
              Offers management
            </h6>
          </paper-mock>
        </grid-mock>
        <grid-mock
          item={true}
          onClick={[Function]}
        >
          <paper-mock
            className="home-view-paper"
          >
            <img
              className="home-menu-icon"
              src="test-file-stub"
            />
            <h6>
              Permission management
            </h6>
          </paper-mock>
        </grid-mock>
      </grid-mock>
    </grid-mock>
  </grid-mock>
  <div
    className="published-system-information"
  >
    <published-system-info-mock />
  </div>
</div>
`;
